packages/salesforcedx-apex-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -514,6 +558,159 @@ export class ApexReplayDebug extends LoggingDebugSession {
     this.sendResponse(response);
   }
 
+  public replaceVariablesWithHeapDump(): void {
+    const topFrame = this.logContext.getTopFrame();
```

Comment:
Based on our discussion the other day to make the replacement generic across different frames, it might be good for this to be in its on class. Also, I think it would be useful to pass the frame in as a parameter (so that it doesn't need to be the topmost frame).